### ISSUE TITLE
VarGFaceNet train from scratch

### ISSUE THREAD

**Issue by qiangcraft** [zma-c-137/VarGFaceNet#5]
When  I use ArcFace to train it from scratch,   'nan' loss value occurs. Could you help me or share some details about how to train it?  And I think the results of y2 and VarGFaceNet which is trained from scratch, are both so good. When I train the y2 with ArcFace, the result is no so good.
![image](https://user-images.githubusercontent.com/18735130/68283719-06b8a200-00b7-11ea-90f7-077d69450885.png)

**Comment by yanmengjia** [comment]
Hi,
y2 is the baseline of Lightweight Face Recognition Challenge & Workshop (ICCV 2019), you can find more details about y2 [here](https://github.com/deepinsight/insightface/tree/master/iccv19-challenge)
You can try to use warmup and cosin learining rate when you find 'nan'. Or you can try to use different initialization methods, xaiver gaussian avg/out mag = 2 is recommended.

**Comment by qianchen94** [comment]
@qiangz520  I also met the problem when the loss is 'nan'. 
How did you slove this problem?
Thank you so much!

**Comment by qiangcraft** [comment]
> @qiangz520 I also met the problem when the loss is 'nan'.
> How did you solve this problem?
> Thank you so much!

@qianchen94 Hello, I solved the problem by decreasing the learning rate or using different initialization methods. As I tried, `Xavier gaussian out mag=2` and `Xavier uniform in mag=2` were both OK. Maybe the main cause was the `learning rate`. 
@yanmengjia And thank you very much.

**Comment by qianchen94** [comment]
@qiangz520 thank you 
when I changed lr from 0.1 to .0.01, it works .
and changed the initalization to Xavier gaussian out mag=2.
By the way, have you got the simlar result about 98% in agedb-30 and cfp-fp?

**Comment by qiangcraft** [comment]
@qianchen94 Not yet, I am also trying to get the best result. Now my best result in those two was about 96.5%.

**Comment by Bryan-bai** [comment]
> 
> 
> > @qiangz520 I also met the problem when the loss is 'nan'.
> > How did you solve this problem?
> > Thank you so much!
> 
> @qianchen94 Hello, I solved the problem by decreasing the learning rate or using different initialization methods. As I tried, `Xavier gaussian out mag=2` and `Xavier uniform in mag=2` were both OK. Maybe the main cause was the `learning rate`.
> @yanmengjia And thank you very much.

 I also met the problem when the loss is 'nan' after decreased the learning rate and used the different initialization methods. Can you share your training process? thx!
